### ISSUE TITLE
Trekstor SurfTab twin 11.6 - ACCEL_MOUNT_MATRIX in hwdb is wrong

### ISSUE THREAD

**Issue by Tintom1** [systemd/systemd#16346]
I'm not sure if there are different versions of this tablet around, but the values of #11771 didn't worked for me. Instead I can confirm this comment: https://github.com/systemd/systemd/commit/d3b8e06deeadae4fc17aa465757690840159aaaf#commitcomment-37423333

For this device the values have to be: ACCEL_MOUNT_MATRIX=-1, 0, 0; 0, 1, 0; 0, 0, -1

Please consider fixing this.

**Comment by mrc0mmand** [comment]
@bedawi could you, please, confirm that this should be the correct matrix value?

**Comment by poettering** [comment]
please file as PR

**Comment by bedawi** [comment]
I do not have the device with me right now. I will reply when I got my hands on it again. Please give me 1-2 weeks.

**Comment by bedawi** [comment]
Hi there. I checked this on my device and I can confirm my initial commit #11771 

@Tintom1 I suggest we compare our full modalias outputs.

```bash
$ cat /sys/`udevadm info -q path -n /dev/iio:device0`/../modalias
acpi:BOSC0200:BOSC0200:
```

```bash
$ cat /sys/class/dmi/id/modalias 
dmi:bvnAmericanMegatrendsInc.:bvrTP15-VT5.3.1.3:bd10/28/2016:svnTrekStor:pnSurfTabtwin11.6:pvrTobefilledbyO.E.M.:rvnHampoo:rnCherryTrailCR:rvrV100:cvnToBeFilledByO.E.M.:ct31:cvrToBeFilledByO.E.M.:
```

On the back of the tablet is written
SurfTab twin 11.6
M.No.: ST11432-2c

Do you have a different output or another model?

@poettering do I have to send another pull request?

**Comment by Tintom1** [comment]
Thanks for replying!
On my device it says:

$ cat /sys/`udevadm info -q path -n /dev/iio:device0`/../modalias
acpi:BOSC0200:BOSC0200:

$ cat /sys/class/dmi/id/modalias
dmi:bvnAmericanMegatrendsInc.:bvrTP15-VT5.2.1.3:bd10/28/2016:svnTrekStor:pnSurfTabtwin11.6:pvrTobefilledbyO.E.M.:rvnHampoo:rnCherryTrailCR:rvrV100:cvnToBeFilledByO.E.M.:ct31:cvrToBeFilledByO.E.M.:


And on the back of the tablet:
SurfTab twin 11.6
KBSTT116-1-DE

**Comment by bedawi** [comment]
Greetings @Tintom1 

Before I send a pull request for this [commit](https://github.com/bedawi/systemd/commit/cefb12e8796ef9151569964d687bc100e219183b) I'd like to ask you to test the new entries on your device. You can either copy these lines into a new file on /etc/udev/hwdb.d/ and run ```systemd-hwdb update``` or to download the [file](https://github.com/bedawi/systemd/blob/cefb12e8796ef9151569964d687bc100e219183b/hwdb.d/60-sensor.hwdb) from my repo to your /etc/udev/hwdb.d/ -folder and run ```systemd-hwdb update```

```bash
$ sudo curl -o /etc/udev/hwdb.d/60-sensor-local.hwdb https://raw.githubusercontent.com/bedawi/systemd/cefb12e8796ef9151569964d687bc100e219183b/hwdb.d/60-sensor.hwdb
$ sudo systemd-hwdb update
$ sudo reboot
```

Please let me know if this fixes the problem for your device.
- Benjamin

**Comment by Tintom1** [comment]
Thanks Benjamin for creating the file.

It works for me now, but I think there might be a new problem. The entries exclude all BIOS-Versions (if there are any) except the two that we're talking about, so the default values are used instead which result in a buggy behavior. I think there should be a catchall and an exception for this device.

The values of my device seem to be exotic compared to other entries that match with string BOSC0200, so I suggest to use the values of your origin commit and check for the exception later. Hence, I modified your file like this:

```diff --git a/60-sensor-local.hwdb b/60-sensor-local.hwdb
index dc9dc5d..ad3dbb3 100644
--- a/60-sensor-local.hwdb
+++ b/60-sensor-local.hwdb
@@ -645,7 +645,7 @@ sensor:modalias:acpi:BMA250*:dmi:*:bvrTREK.G.WI71C.JGBMRBA*:*:svnTrekStor:pnSurf
 
 sensor:modalias:acpi:KIOX000A*:dmi:*:svnTREKSTOR*:pnPrimetabS11B:*
 sensor:modalias:acpi:KIOX000A*:dmi:*:svnTREKSTOR:pnPrimetabT13B:*
-sensor:modalias:acpi:BOSC0200*:dmi:*:bvrTP15-VT5.3.1.3:*:svnTrekStor*:pnSurfTabtwin11.6:*
+sensor:modalias:acpi:BOSC0200*:dmi:*:svnTrekStor*:pnSurfTabtwin11.6:*
  ACCEL_MOUNT_MATRIX=1, 0, 0; 0, -1, 0; 0, 0, 1
 
 sensor:modalias:acpi:BOSC0200*:dmi:*:bvrTP15-VT5.2.1.3:*:svnTrekStor*:pnSurfTabtwin11.6:*
```


This works for me as well.
Tino

**Comment by bedawi** [comment]
I created a new pull request to deal with this issue:
https://github.com/systemd/systemd/pull/16615

**Comment by claell** [comment]
I also own this device and came across this issue since I am facing the problem that the sensor values seem off by 180 degree for me, tested on a Fedora live system (should be Fedora 32).

I had some short testing with adding a  hwdb file and `systemd-hwdb update` but it did not work with either of the proposed matrices for the device. I guess the file I created had problems and was not applied? I just wanted to make sure it will work out of the box for my device in the future. Can you post an easy to apply workflow to test, maybe using a broader wildcard since I just want to try it on my specific device?

**Comment by bedawi** [comment]
Hi @claell - usually a "reboot" does the trick, but do not forget '''sudo systemd-hwdb update''' before reboot.

If you happen to have another version of the device then I'd interested in the outputs of 
```$ cat /sys/`udevadm info -q path -n /dev/iio:device0`/../modalias```
and 
```$ cat /sys/class/dmi/id/modalias ```

Good luck!

**Comment by claell** [comment]
Alright. I thought that command would only temporarily apply the changes and reset after reboot.

I only noticed that on a live image (Fedora), I have not installed a Linux distro on that device at the moment. Thus a reboot resets the settings for me. I just wanted to contribute so this device gets better support, although I might as well never really use a Linux distribution on it. Is there any other way with low effort involved?

Also I noticed that with a Debian live image the orientation worked properly, maybe because they use an older version of the hwdb.
